src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -603,14 +612,73 @@ private FingerprintIdDetails createUnknownFP(final Path samFile, final SAMRecord
+        }
         final Map<FingerprintIdDetails, Fingerprint> fpIdDetailsMap = this.fingerprintSamFile(samFile, h -> new HaplotypeProbabilitiesFromContaminatorSequence(h, contamination));
+        final Map<String, Fingerprint> fpIdDetailsBySample = Fingerprint.mergeFingerprintsBy(fpIdDetailsMap,
+                Fingerprint.getFingerprintIdDetailsStringFunction(CrosscheckMetric.DataType.SAMPLE)).entrySet().stream()
+                .collect(Collectors.toMap(e -> e.getKey().sample, Map.Entry::getValue));
 
-        final Map<FingerprintIdDetails, Fingerprint> fpIdDetailsBySample = Fingerprint.mergeFingerprintsBy(fpIdDetailsMap,
-                Fingerprint.getFingerprintIdDetailsStringFunction(CrosscheckMetric.DataType.SAMPLE));
+        if (backgroundFPFile != null) {
+            final Map<FingerprintIdDetails, Fingerprint> fpIdDetailsMapBackground = this.fingerprintFiles(Collections.singleton(backgroundFPFile), 1, 1, TimeUnit.DAYS);
+            final Map<FingerprintIdDetails, Fingerprint> fpIdDetailsMapBackgroundCorrespondingSample = new HashMap<>();
+            for(final FingerprintIdDetails fingerprintIdDetails : fpIdDetailsMapBackground.keySet()) {
+                final String correspondingSample = sampleCorrespondenceMap.get(fingerprintIdDetails.sample);
+                if (correspondingSample != null) {
+                    final FingerprintIdDetails fingerprintIdDetailsCorrespondingSample = new FingerprintIdDetails(fingerprintIdDetails.platformUnit, backgroundFPFile.toUri().toString());
+                    fingerprintIdDetailsCorrespondingSample.sample = correspondingSample;
+                    fingerprintIdDetailsCorrespondingSample.library = fingerprintIdDetails.library;
+
+                    fpIdDetailsMapBackgroundCorrespondingSample.put(fingerprintIdDetailsCorrespondingSample, fpIdDetailsMapBackground.get(fingerprintIdDetails));
+                }
+            }
+            final Map<String, Fingerprint> fpIdDetailsBySampleBackground = Fingerprint.mergeFingerprintsBy(fpIdDetailsMapBackgroundCorrespondingSample,
+                    Fingerprint.getFingerprintIdDetailsStringFunction(CrosscheckMetric.DataType.SAMPLE)).entrySet().stream()
+                    .collect(Collectors.toMap(e -> e.getKey().sample, Map.Entry::getValue));
+
+            //merge background fingerprint into contaminated fingerprint
+            for (final Map.Entry<String, Fingerprint> entry : fpIdDetailsBySampleBackground.entrySet()) {
+                final String sample = entry.getKey();
+                final Fingerprint backgroundFingerprint = entry.getValue();
+                final Fingerprint contaminatedFingerprint = fpIdDetailsBySample.get(sample);
+                for (final Map.Entry<HaplotypeBlock, HaplotypeProbabilities> fingerprintEntry : backgroundFingerprint.entrySet()) {
+                    if (contaminatedIsTumor && !isConfidentlyHom(fingerprintEntry.getValue())) {
```

Comment:
this will not work well in the cases where the fingerprints are not confident...is there a better way to do this that is more Bayesian?